File: frontend/scripts/domain/Room.ts
import { ChatRoomMessage } from './ChatRoomMessage';
import { User } from './User';

export class Room {

  public static fromPOJOMapToList(roomMap: Map<string, Room>): Array<Room> {
    let rooms: Array<Room> = new Array<Room>();

    for(let roomName in roomMap) {
      let roomPOJO = roomMap[roomName];
      let room: Room = Room.fromPOJO(roomPOJO);
      rooms.push(room);
    }

    return rooms;
  }

  public static fromPOJO(pojo: Object) {
    let room: Room = new Room();

    room.name = pojo['name'];
    room.maxNumberOfUsers = pojo['maxNumberOfUsers'];
    room.totalNumberOfUsers = pojo['totalNumberOfUsers'];
    room.messages = Room.messagesFromPOJOList(pojo['messages']);
    room.users = Room.usersFromPOJOList(pojo['users']);

    return room;
  }

  private static messagesFromPOJOList(messagePOJOs: Array<Object>): Array<ChatRoomMessage> {
    let messages = new Array<ChatRoomMessage>();

    messagePOJOs.forEach((messagePOJO: Object) => {
      messages.push(ChatRoomMessage.fromPOJO(messagePOJO));
    });

    return messages;
  }

  private static usersFromPOJOList(usersMap: Object): Array<User> {
    let users = new Array<User>();

    for(let name in usersMap) {
      let userPOJO: Object = usersMap[name];
      users.push(User.fromPOJO(userPOJO));
    }

    return users;
  }

  private name: string;
  private maxNumberOfUsers: number;
  private totalNumberOfUsers: number;
  private messages: Array<ChatRoomMessage>;
  private users: Array<User>;

  public setName(name: string): void {
    this.name = name;
  }

  public getName(): string {
    return this.name;
  }

  public getMaxNumberOfUsers(): number {
    return this.maxNumberOfUsers;
  }

  public getTotalNumberOfUsers(): number {
    return this.totalNumberOfUsers;
  }

  public getMessages(): Array<ChatRoomMessage> {
    return this.messages;
  }

  public getUsers():  Array<User> {
    return this.users;
  }

}

File: src/main/java/l2k/demo/multiple/chats/domain/Room.java
package l2k.demo.multiple.chats.domain;

import java.security.Principal;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

public class Room {
	
	private static final int MAXIMUM_NUMBER_OF_PERSISTED_MESSAGES = 75;
	
	private String name;
	private Map<String, Principal> users;
	private int maxNumberOfUsers;
	
	private LinkedList<ChatRoomMessage> messages;
	
	{
		setUsers(new HashMap<String, Principal>());
		setMessages(new LinkedList<ChatRoomMessage>());
	}
	
	public boolean isFull() {
		return getUsers().size() >= maxNumberOfUsers;
	}
	
	public void addMessage(ChatRoomMessage message) {
		if(getMessages().size() >= MAXIMUM_NUMBER_OF_PERSISTED_MESSAGES) {
			getMessages().removeFirst();
		}
		
		getMessages().addLast(message);
	}
	
	public void addUser(Principal user) {
		getUsers().put(user.getName(), user);
	}
	
	public boolean contains(User user) {
		return getUsers().get(user.getName()) != null;
	}
	
	public int getTotalNumberOfUsers() {
		return getUsers().size();
	}
	
	public void removeUser(Principal user) {
		getUsers().remove(user.getName());
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public int getMaxNumberOfUsers() {
		return maxNumberOfUsers;
	}

	public void setMaxNumberOfUsers(int maxNumberOfUsers) {
		this.maxNumberOfUsers = maxNumberOfUsers;
	}

	public LinkedList<ChatRoomMessage> getMessages() {
		return messages;
	}

	public void setMessages(LinkedList<ChatRoomMessage> messages) {
		this.messages = messages;
	}

	public Map<String, Principal> getUsers() {
		return users;
	}

	public void setUsers(Map<String, Principal> users) {
		this.users = users;
	}
	
}
